File: CodingTestPlatform/BOJ/1_Bronze/2/3040/3040.py
# 백설 공주와 일곱 난쟁이
# https://www.acmicpc.net/problem/3040

from itertools import combinations
import sys
input = sys.stdin.readline
for i in combinations([int(input()) for _ in range(9)], 7):
    if sum(i) == 100:
        for j in sorted(i):
            print(j)
        break

File: CodingTestPlatform/BOJ/1_Bronze/2/3040/3040.js
/**
 * 백설 공주와 일곱 난쟁이 - 2309 일곱 난쟁이 와 동일
 * Bronze2
 * https://www.acmicpc.net/problem/3040
 */

 const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');
 const numArr = input.map((str) => parseInt(str));
 const sum = numArr.reduce((acc, num) => acc + num, 0);
 
 let result;
 
 for (let i = 0; i < numArr.length - 1; i++) {
     for (let j = i + 1; j < numArr.length; j++) {
         if (sum - numArr[i] - numArr[j] === 100) {
            result = numArr.filter((num) => (num !== numArr[i] && num !== numArr[j])); 
            break;
         }
     }
 }
 
 console.log(result.sort((a, b) => a - b).join('\n'));